B/BOLT/build_tarballs.jl

```diff
@@ -0,0 +1,169 @@
+CMAKE_FLAGS+=(-DCMAKE_CROSSCOMPILING=True)
+
+# Tell LLVM where our pre-built tblgen tools are
+CMAKE_FLAGS+=(-DLLVM_TABLEGEN=${WORKSPACE}/bootstrap/bin/llvm-tblgen)
+CMAKE_FLAGS+=(-DLLVM_CONFIG_PATH=${WORKSPACE}/bootstrap/bin/llvm-config)
+
+# Explicitly use our cmake toolchain file
+CMAKE_FLAGS+=(-DCMAKE_TOOLCHAIN_FILE=${CMAKE_TARGET_TOOLCHAIN})
+
+# Manually set the host triplet, as otherwise on some platforms it tries to guess using
+# `ld -v`, which is hilariously wrong.
+CMAKE_FLAGS+=(-DLLVM_HOST_TRIPLE=${target})
+
+# Most targets use the actual target string, but we disagree on `aarch64-darwin` and `arm64-darwin`
+CMAKE_TARGET=${target}
+
+if [[ "${target}" == *apple* ]]; then
+    # On OSX, we need to override LLVM's looking around for our SDK
+    # We need to link against libc++ on OSX
+    CMAKE_FLAGS+=(-DLLVM_ENABLE_LIBCXX=ON)
+
+    # If we're building for Apple, CMake gets confused with `aarch64-apple-darwin` and instead prefers
+    # `arm64-apple-darwin`.  If this issue persists, we may have to change our triplet printing.
+    if [[ "${target}" == aarch64* ]]; then
+        CMAKE_TARGET=arm64-${target#*-}
+    fi
+fi
+
+GCC_VERSION=$(gcc --version | head -1 | awk '{ print $3 }' | cut -d. -f1)
+if [[ $version -le 10 && "${target}" == aarch64-linux* ]]; then
+    CMAKE_C_FLAGS+=(-mno-outline-atomics)
+    CMAKE_CPP_FLAGS+=(-mno-outline-atomics)
+fi
+
+# Tell LLVM which compiler target to use, because it loses track for some reason
+CMAKE_FLAGS+=(-DCMAKE_C_COMPILER_TARGET=${CMAKE_TARGET})
+CMAKE_FLAGS+=(-DCMAKE_CXX_COMPILER_TARGET=${CMAKE_TARGET})
+CMAKE_FLAGS+=(-DCMAKE_ASM_COMPILER_TARGET=${CMAKE_TARGET})
+
+# Defaults to off when crosscompiling from LLVM 18
+CMAKE_FLAGS+=(-DBOLT_ENABLE_RUNTIME=true)
+
+cmake -GNinja ${LLVM_SRCDIR} ${CMAKE_FLAGS[@]} -DCMAKE_CXX_FLAGS=\"${CMAKE_CPP_FLAGS[*]} ${CMAKE_CXX_FLAGS[*]}\" -DCMAKE_C_FLAGS=\"${CMAKE_C_FLAGS[*]}\"
+ninja -j${nproc} -vv bolt
+
+# Install!
+ninja install-llvm-bolt
+
+install_license ${WORKSPACE}/srcdir/llvm-project/bolt/LICENSE.TXT
+"""
+
+sources = BinaryBuilder.AbstractSource[
+    GitSource("https://github.com/llvm/llvm-project.git", git_sha),
+    ArchiveSource(
+        "https://github.com/phracker/MacOSX-SDKs/releases/download/10.15/MacOSX10.14.sdk.tar.xz",
+        "0f03869f72df8705b832910517b47dd5b79eb4e160512602f593ed243b28715f"
+    )
+]
+
+platforms = expand_cxxstring_abis(supported_platforms())
+filter!(p -> arch(p) ∈ ("x86_64", "aarch64") && os(p) ∈ ("linux", "macos"), platforms)
+
+products = [
+    ExecutableProduct("llvm-bolt", :llvm_bolt),
+    ExecutableProduct("merge-fdata", :merge_fdata)
+]
+
+name = "BOLT"
+
+# Dependencies that must be installed before this package can be built
+# TODO: LibXML2
+dependencies = [
+    Dependency("Zlib_jll"), # for LLD&LTO
+]
+
+build_tarballs(ARGS, name, version, sources, buildscript, platforms, products, dependencies;
+               preferred_gcc_version=v"10", preferred_llvm_version=v"16", julia_compat="1.12")
```

Comment:
Does this really require julia v1.12?  Since this is just an external executable I think this should work with any julia version (well, starting from v1.6)
```suggestion
               preferred_gcc_version=v"10", preferred_llvm_version=v"16", julia_compat="1.6")
```